diff -burN gcc-8.2.0.orig/config.sub gcc-8.2.0/config.sub
--- gcc-8.2.0.orig/config.sub	2018-01-02 20:25:18.000000000 -0800
+++ gcc-8.2.0/config.sub	2018-08-02 23:30:53.000000000 -0700
@@ -778,6 +778,10 @@
 		basic_machine=i386-unknown
 		os=-vsta
 		;;
+	iop)
+		basic_machine=mipsel-scei
+		os=-irx
+		;;
 	iris | iris4d)
 		basic_machine=mips-sgi
 		case $os in
@@ -1416,7 +1420,8 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
+	      | -irx*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -burN gcc-8.2.0.orig/configure gcc-8.2.0/configure
--- gcc-8.2.0.orig/configure	2018-07-26 02:48:58.000000000 -0700
+++ gcc-8.2.0/configure	2018-08-02 23:30:08.000000000 -0700
@@ -3825,6 +3825,8 @@
     ;;
   mips*-*-linux*)
     ;;
+  mipsel-*-irx*)
+    ;;
   mips*-*-ultrix* | mips*-*-osf* | mips*-*-ecoff* | mips*-*-pe* \
   | mips*-*-irix* | mips*-*-lnews* | mips*-*-riscos*)
     noconfigdirs="$noconfigdirs ld gas gprof"
diff -burN gcc-8.2.0.orig/configure.ac gcc-8.2.0/configure.ac
--- gcc-8.2.0.orig/configure.ac	2018-02-13 05:01:33.000000000 -0800
+++ gcc-8.2.0/configure.ac	2018-08-02 23:30:09.000000000 -0700
@@ -1156,6 +1156,8 @@
     ;;
   mips*-*-linux*)
     ;;
+  mipsel-*-irx*)
+    ;;
   mips*-*-ultrix* | mips*-*-osf* | mips*-*-ecoff* | mips*-*-pe* \
   | mips*-*-irix* | mips*-*-lnews* | mips*-*-riscos*)
     noconfigdirs="$noconfigdirs ld gas gprof"
diff -burN gcc-8.2.0.orig/gcc/config.gcc gcc-8.2.0/gcc/config.gcc
--- gcc-8.2.0.orig/gcc/config.gcc	2018-06-25 12:34:01.000000000 -0700
+++ gcc-8.2.0/gcc/config.gcc	2018-08-02 23:30:09.000000000 -0700
@@ -2288,6 +2288,11 @@
 	tmake_file="mips/t-elf"
 	tm_defines="${tm_defines} MIPS_ISA_DEFAULT=3 MIPS_ABI_DEFAULT=ABI_N32"
 	;;
+mipsel-*-irx*)
+	tm_file="dbxelf.h elfos.h newlib-stdint.h ${tm_file} mips/elf.h"
+	tmake_file="mips/t-elf mips/dbx.h"
+	target_cpu_default=MASK_SOFT_FLOAT_ABI
+	;;
 mips64-*-elf* | mips64el-*-elf*)
 	tm_file="elfos.h newlib-stdint.h ${tm_file} mips/elf.h"
 	tmake_file="mips/t-elf"
diff -burN gcc-8.2.0.orig/libgcc/config.host gcc-8.2.0/libgcc/config.host
--- gcc-8.2.0.orig/libgcc/config.host	2018-04-06 13:04:17.000000000 -0700
+++ gcc-8.2.0/libgcc/config.host	2018-08-02 23:30:09.000000000 -0700
@@ -930,6 +930,9 @@
 	tmake_file="$tmake_file mips/t-elf mips/t-crtstuff mips/t-mips16"
 	extra_parts="$extra_parts crti.o crtn.o"
 	;;
+mipsel-*-irx)
+	tmake_file="$tmake_file mips/t-elf"
+	;;
 mipsr5900-*-elf* | mipsr5900el-*-elf*)
 	tmake_file="$tmake_file mips/t-elf mips/t-crtstuff"
 	extra_parts="$extra_parts crti.o crtn.o"
